### PR TITLE
Add redeployment and external-access docs

### PR DESCRIPTION
**Pull request by weiiwang01**
Add two new documents: the redeployment guide and the external access Requirements.

The redeployment guide describes the steps users need to follow to migrate from an existing WordPress deployment to a new charm deployment. The external access requirements document details the necessary internet access for WordPress to function properly after deployment.

### CODE REVIEW

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   integration-test:
     name: Integration Test
-    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
```
Comment (arturo-seijas) [inline .github/workflows/integration_test.yaml:10]: IS this intended?

File: .github/workflows/integration_test_juju3.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   integration-test-juju3:
     name: Integration Test on Juju3
-    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
```
Comment (arturo-seijas) [inline .github/workflows/integration_test_juju3.yaml:10]: Same here

File: docs/how-to/redeployment.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Redeployment Guide
+
+This guide provides the steps necessary for migrating an existing WordPress
+instance to a new charm instance.
+
+## Migrate Database
+
+Follow the instructions
+in [the mysql charm migration guide](https://charmhub.io/mysql/docs/h-migrate-cluster-via-restore)
```
Comment (yanksyoon) [inline docs/how-to/redeployment.md:9]: I thought `h-migrate-cluster-via-restore` was a typo but it seems it isn't :joy:

File: docs/how-to/redeployment.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Redeployment Guide
+
+This guide provides the steps necessary for migrating an existing WordPress
```
Comment (arturo-seijas) [inline docs/how-to/redeployment.md:3]: s/steps necessary/necessary steps?

**Review by yanksyoon — DISMISSED** [review]
LGTM!

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (weiiwang01) [reply .github/workflows/integration_test.yaml:10]: No, I am just using this pull request to test the new operator-workflow pipeline, I will revert this after I saw the result, thanks!

File: .github/workflows/integration_test_juju3.yaml (hunk 2 above)
Comment (weiiwang01) [reply .github/workflows/integration_test_juju3.yaml:10]: see comment above.

File: docs/how-to/redeployment.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+## Migrate Media Files
+
+### Media Files Stored in Kubernetes Storage
+
+If your media files are stored in Kubernetes
+storage (`wp_plugin_openstack-objectstorage_config` is not configured), use the
+following steps to migrate your files:
+
+1. Use the `kubectl cp` command to transfer files from
```
Comment (amandahla) [inline docs/how-to/redeployment.md:20]: juju scp wouldnt work?

Like:
https://discourse.canonical.com/t/increase-storage-used-by-a-charm-workaround/2760/5?u=amandahla

File: docs/reference/external-access.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+## OpenStack Object Storage
+
+When activated, the WordPress instance will need to connect to configured
+OpenStack object storage for uploading and retrieval of media and assets.
+
+## Akismet Spam Protection Plugin
+
+The Akismet spam protection plugin, when enabled via the `wp_plugin_akismet_key`
+configuration, requires internet access to connect with the Akismet API server.
```
Comment (amandahla) [inline docs/reference/external-access.md:14]: Is there a URL or reference for this access?

File: docs/how-to/redeployment.md (hunk 5 above)
Comment (weiiwang01) [reply docs/how-to/redeployment.md:20]: Yes, juju scp would be a better choice here, thanks!

File: docs/how-to/redeployment.md (hunk 4 above)
Comment (weiiwang01) [reply docs/how-to/redeployment.md:3]: Update, thanks!

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (weiiwang01) [reply .github/workflows/integration_test.yaml]: It was a success, removed.

File: docs/how-to/redeployment.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# Redeployment Guide
```
Comment (varshigupta12) [inline docs/how-to/redeployment.md:1]: I think this should be renamed as "Redploy". So the navigation reads as "How-to"-> "Redeploy"

Comment (mthaddon) [reply docs/how-to/redeployment.md:1]: Do you mean the filename itself (currently `redeployment.md`) or the title here (or both)?

Comment (weiiwang01) [reply docs/how-to/redeployment.md]: Yes, indeed, updated, thanks!

File: docs/how-to/redeploy.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Redeploy
```
Comment (arturo-seijas) [inline docs/how-to/redeploy.md:1]: How to redeploy

Comment (weiiwang01) [reply docs/how-to/redeploy.md:1]: Updated, thanks!

**Review by arturo-seijas — APPROVED** [review]

**Review by mthaddon — APPROVED** [review]

**Pull request merged**